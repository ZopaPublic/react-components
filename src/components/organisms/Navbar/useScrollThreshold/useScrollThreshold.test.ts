import { renderHook } from '@testing-library/react-hooks';
import useScrollThreshold from './useScrollThreshold';

describe('HOOK: useScrollThreshold', () => {
  let origAddEventListener: typeof global.document.addEventListener;
  let origScrollY: number;
  let origRAF: typeof global.requestAnimationFrame;

  beforeEach(() => {
    origAddEventListener = global.document.addEventListener;
    origRAF = global.requestAnimationFrame;
    origScrollY = global.scrollY;

    global.document.addEventListener = jest.fn();
    global.requestAnimationFrame = jest.fn();
  });

  afterEach(() => {
    global.document.addEventListener = origAddEventListener;
    global.requestAnimationFrame = origRAF;
    global.scrollY! = origScrollY;
  });

  it('registers a scroll listener when called', () => {
    renderHook(useScrollThreshold);

    expect(global.document.addEventListener).toHaveBeenCalledTimes(1);
    expect((global.document.addEventListener as jest.Mock).mock.calls[0]).toMatchSnapshot();
  });

  it.each`
    scrollY | threshold | isAbove
    ${0}    | ${10}     | ${false}
    ${20}   | ${10}     | ${true}
    ${10}   | ${10}     | ${false}
  `(
    'user scrolled:  $scrollY, threshold was: $threshold, notifies above: $isAbove',
    ({ scrollY, threshold, isAbove }) => {
      global.scrollY! = scrollY;

      const {
        result: { current: isOverTheTreshold },
      } = renderHook(() => useScrollThreshold(threshold));

      expect(isOverTheTreshold).toBe(isAbove);
    },
  );
});
