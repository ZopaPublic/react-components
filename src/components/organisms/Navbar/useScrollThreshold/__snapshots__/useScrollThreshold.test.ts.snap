// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOOK: useScrollThreshold registers a scroll listener when called 1`] = `
Array [
  "selectionchange",
  [Function],
  false,
]
`;
